### PR TITLE
Solving SonarQube issues

### PR DESCRIPTION
**Pull request by mvanOsch**

### CODE REVIEW

**Review by MithraicMagic — APPROVED** [review]
Looks good!

File: osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringManagementEndpoint.java [hunk 1]
```diff
@@ -152,8 +155,8 @@ public FindEventsResponse getFindEventsResponse(@OrganisationIdentification fina
 
         } catch (final ConstraintViolationException e) {
             LOGGER.error("FindEventsRequest Exception", e.getMessage(), e.getStackTrace(), e);
-            throw new FunctionalException(FunctionalExceptionType.VALIDATION_ERROR, ComponentType.WS_SMART_METERING,
-                    new ValidationException(e.getConstraintViolations()));
+            this.handleException(new FunctionalException(FunctionalExceptionType.VALIDATION_ERROR, ComponentType.WS_SMART_METERING,
+                    new ValidationException(e.getConstraintViolations())));
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringManagementEndpoint.java:159]: see DeviceManagementEndpoint

File: osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringManagementEndpoint.java [hunk 2]
```diff
@@ -221,6 +224,7 @@ public EnableDebuggingAsyncResponse enableDebuggingRequest(
         return response;
     }
 
+    @SuppressWarnings("squid:S3457")
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringManagementEndpoint.java:227]: Explanation is missing.

File: osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringManagementEndpoint.java [hunk 3]
```diff
@@ -285,6 +289,7 @@ public DisableDebuggingAsyncResponse disableDebuggingRequest(
...
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringManagementEndpoint.java:292]: Explanation is missing

File: osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringManagementEndpoint.java [hunk 4]
```diff
@@ -369,6 +374,7 @@ public FindMessageLogsAsyncResponse findMessageLogsRequest(
      * @return FindMessageLogsResponse
      * @throws OsgpException
      */
+    @SuppressWarnings("squid:S3457")
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringManagementEndpoint.java:377]: Explanation is missing

File: osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringManagementEndpoint.java [hunk 5]
```diff
@@ -436,6 +442,7 @@ public SetDeviceCommunicationSettingsAsyncResponse setDeviceCommunicationSetting
         return response;
     }
 
+    @SuppressWarnings("squid:S3457")
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringManagementEndpoint.java:445]: Explanation is missing

File: osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringManagementEndpoint.java [hunk 6]
```diff
@@ -501,6 +508,7 @@ public SetDeviceLifecycleStatusByChannelAsyncResponse setDeviceLifecycleStatusBy
         return asyncResponse;
     }
 
+    @SuppressWarnings("squid:S3457")
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringManagementEndpoint.java:511]: Explanation is missing

File: osgp/platform/osgp-adapter-ws-tariffswitching/src/main/java/org/opensmartgridplatform/adapter/ws/tariffswitching/endpoints/TariffSwitchingAdHocManagementEndpoint.java [hunk 7]
```diff
@@ -81,8 +81,8 @@ public GetDevicesResponse getDevices(@OrganisationIdentification final String or
             response.setDevicePage(devicePage);
         } catch (final ConstraintViolationException e) {
             LOGGER.error("Exception: {}, StackTrace: {}", e.getMessage(), e.getStackTrace(), e);
-            throw new FunctionalException(FunctionalExceptionType.VALIDATION_ERROR, COMPONENT_WS_TARIFF_SWITCHING,
-                    new ValidationException(e.getConstraintViolations()));
+            this.handleException(new FunctionalException(FunctionalExceptionType.VALIDATION_ERROR, COMPONENT_WS_TARIFF_SWITCHING,
+                    new ValidationException(e.getConstraintViolations())));
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-ws-tariffswitching/src/main/java/org/opensmartgridplatform/adapter/ws/tariffswitching/endpoints/TariffSwitchingAdHocManagementEndpoint.java:85]: see DeviceManagementEndpoint

File: osgp/platform/osgp-adapter-ws-tariffswitching/src/main/java/org/opensmartgridplatform/adapter/ws/tariffswitching/endpoints/TariffSwitchingScheduleManagementEndpoint.java [hunk 8]
```diff
@@ -99,8 +99,9 @@ public SetScheduleAsyncResponse setSchedule(@OrganisationIdentification final St
             response.setAsyncResponse(asyncResponse);
         } catch (final ConstraintViolationException e) {
             LOGGER.error("Exception: {}, StackTrace: {}", e.getMessage(), e.getStackTrace(), e);
-            throw new FunctionalException(FunctionalExceptionType.VALIDATION_ERROR, ComponentType.WS_TARIFF_SWITCHING,
-                    new ValidationException(e.getConstraintViolations()));
+            this.handleException(new FunctionalException(FunctionalExceptionType.VALIDATION_ERROR,
+                ComponentType.WS_TARIFF_SWITCHING,
+                    new ValidationException(e.getConstraintViolations())));
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-ws-tariffswitching/src/main/java/org/opensmartgridplatform/adapter/ws/tariffswitching/endpoints/TariffSwitchingScheduleManagementEndpoint.java:104]: see DeviceManagementEndpoint

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/entities/Ssld.java [hunk 9]
```diff
@@ -97,6 +97,7 @@ public Ssld(final String deviceIdentification, final String alias, final Address
         super(deviceIdentification, alias, containerAddress, gpsCoordinates, cdmaSettings);
     }
 
+    @SuppressWarnings("squid:S2159")
```
Comment (robindenadel) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/entities/Ssld.java:100]: Explanation is missing

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/exceptions/ValidationException.java [hunk 10]
```diff
@@ -16,6 +16,9 @@
 import org.springframework.ws.soap.server.endpoint.annotation.FaultCode;
 import org.springframework.ws.soap.server.endpoint.annotation.SoapFault;
 
+// suppress warning about the Set<? extends ConstraintViolation<?>>. This is because it is apparently required to be
+// written this way.
+@SuppressWarnings("squid:S1452")
```
Comment (robindenadel) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/exceptions/ValidationException.java:21]: I don't consider this a good explanation.

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/AbstractPushSetup.java [hunk 11]
```diff
@@ -152,6 +152,10 @@ public boolean hasPushObjectList() {
         return this.pushObjectList != null;
     }
 
+    // suppress warning about returning null. With this method a null return is currently used to show that the list
+    // that is used isn't actually initialized. This means that there isn't really an absence of objects to put in
+    // the list, but rather that the object to put into it couldn't be made.
+    @SuppressWarnings("squid:S1168")
```
Comment (robindenadel) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/AbstractPushSetup.java:158]: I don't think we should suppress this warning, and return an empty list.
Write unit tests for all the code calling this method to make sure everything keeps on working ok.

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/AbstractPushSetup.java [hunk 12]
```diff
@@ -171,6 +175,10 @@ public boolean hasCommunicationWindow() {
         return this.communicationWindow != null;
     }
 
+    // suppress warning about returning null. With this method a null return is currently used to show that the list
+    // that is used isn't actually initialized. This means that there isn't really an absence of objects to put in
+    // the list, but rather that the object to put into it couldn't be made.
+    @SuppressWarnings("squid:S1168")
```
Comment (robindenadel) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/AbstractPushSetup.java:181]: I don't think we should suppress this warning, and return an empty list.
Write unit tests for all the code calling this method to make sure everything keeps on working ok.

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/ClockStatus.java [hunk 13]
```diff
@@ -51,6 +51,9 @@ public boolean hasStatusBits() {
         return this.statusBits != null;
     }
 
+    // this is a getter, meaning it should return what the actual value is. Since it is possible that the actual
+    // value is set to null, it should also return this in that case.
+    @SuppressWarnings("squid:S1168")
```
Comment (robindenadel) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/ClockStatus.java:56]: I don't think we should suppress this warning, and return an empty set.
Write unit tests for all the code calling this method to make sure everything keeps on working ok.

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/ClockStatusBit.java [hunk 14]
```diff
@@ -45,6 +48,10 @@ public String toString() {
         return forClockStatus(clockStatus & 0xFF);
     }
 
+    // here returning null is independed of the actual ClockStatusBit that is later used to make the list. This meand
+    // that returning an empty list would falsly suggest that there is an empty list, where in fact there is no valid
+    // list to return
+    @SuppressWarnings("squid:S1168")
```
Comment (robindenadel) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/ClockStatusBit.java:54]: I don't think we should suppress this warning, and return an empty set.
Write unit tests for all the code calling this method to make sure everything keeps on working ok.

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/SpecificAttributeValueRequest.java [hunk 15]
```diff
@@ -33,6 +33,10 @@ public int hashCode() {
         return result;
     }
 
+    // Suppress warning that super.equals(obj) always returns false. This is not true since the given object might
+    // be the parent class from this class. I'm not sure if the current behaviour is intended but that should be
+    // tested in unit tests.
+    @SuppressWarnings("squid:S2159")
```
Comment (robindenadel) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/SpecificAttributeValueRequest.java:39]: Don't suppress this warning

File: osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/infra/networking/services/commands/Iec61850GetConfigurationCommand.java [hunk 16]
```diff
@@ -41,6 +41,9 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+// small piece of code for making softwareConfiguration and lightTypeValue also exists in GetStatusCommand. The code
+// is too small to be put in a seperate class. Doing so also wouldn't help with making the code easier to maintain.
+@SuppressWarnings("DuplicatedCode")
```
Comment (robindenadel) [inline osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/infra/networking/services/commands/Iec61850GetConfigurationCommand.java:46]: Don't suppress this warning

**Review by robindenadel — CHANGES_REQUESTED** [review]
IDE settings for formatting and save actions are not OK.
See https://documentation.opensmartgridplatform.org/Opensourcecommunity/ToolsguidelinesCI.html

**Comment by mvanOsch** [comment]
retest this please

**Comment by mvanOsch** [comment]
retest this please

**Comment by mvanOsch** [comment]
retest this please

File: osgp/platform/osgp-adapter-ws-core/src/test/java/org/opensmartgridplatform/adapter/ws/core/application/services/DeviceManagementServiceTest.java [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+        when(this.eventSpecifications.withDescription(any())).thenReturn(this.descriptionSpecification);
+        when(this.pagingSettings.getPageNumber()).thenReturn(1);
+        when(this.pagingSettings.getPageSize()).thenReturn(1);
+        when(this.eventRepository.findAll((Specification<Event>) any(), any(Pageable.class))).thenReturn(null);
+        when(this.specification.and(any())).thenReturn(this.specification);
+        when(this.descriptionSpecification.or(any())).thenReturn(this.descriptionSpecification);
+
+        final Page<Event> resultPage = this.deviceManagementService.findEvents(this.criteria);
+
+        Assertions.assertThat(resultPage).isNull();
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-ws-core/src/test/java/org/opensmartgridplatform/adapter/ws/core/application/services/DeviceManagementServiceTest.java:71]: use a static import so we don't need to put Assertions. here

File: osgp/platform/osgp-adapter-ws-core/src/test/java/org/opensmartgridplatform/adapter/ws/core/application/services/DeviceManagementServiceTest.java [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+        doNothing().when(this.domainHelperService).isAllowed(any(), any(), any());
+        when(this.eventSpecifications.isFromDevice(any())).thenReturn(new Specification<Event>() {
+            private static final long serialVersionUID = 2946693984484298490L;
+
+            @Override
+            public Predicate toPredicate(final Root<Event> root, final CriteriaQuery<?> criteriaQuery,
+                    final CriteriaBuilder criteriaBuilder) {
+                return null;
+            }
+        });
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-ws-core/src/test/java/org/opensmartgridplatform/adapter/ws/core/application/services/DeviceManagementServiceTest.java:90]: refactor the return part into a method

File: osgp/platform/osgp-adapter-ws-core/src/test/java/org/opensmartgridplatform/adapter/ws/core/application/services/DeviceManagementServiceTest.java [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+import org.opensmartgridplatform.domain.core.repositories.EventRepository;
+import org.opensmartgridplatform.domain.core.specifications.EventSpecifications;
+import org.opensmartgridplatform.shared.application.config.PagingSettings;
+import org.opensmartgridplatform.shared.exceptionhandling.FunctionalException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+@ExtendWith(MockitoExtension.class)
+class DeviceManagementServiceTest {
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-ws-core/src/test/java/org/opensmartgridplatform/adapter/ws/core/application/services/DeviceManagementServiceTest.java:31]: there are only tests checking for the resultPage to be null
I would expect tests checking a non-null result as well

File: osgp/platform/osgp-adapter-ws-core/src/test/java/org/opensmartgridplatform/adapter/ws/core/application/services/DeviceManagementServiceTest.java [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+        doNothing().when(this.domainHelperService).isAllowed(any(), any(), any());
+        when(this.eventSpecifications.isFromDevice(any())).thenReturn(new Specification<Event>() {
+            private static final long serialVersionUID = 2946693984484298490L;
+
+            @Override
+            public Predicate toPredicate(final Root<Event> root, final CriteriaQuery<?> criteriaQuery,
+                    final CriteriaBuilder criteriaBuilder) {
+                return null;
+            }
+        });
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-ws-core/src/test/java/org/opensmartgridplatform/adapter/ws/core/application/services/DeviceManagementServiceTest.java:117]: see earlier comment

File: osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringManagementEndpoint.java [hunk 21]
```diff
@@ -68,7 +68,6 @@
 import org.springframework.ws.server.endpoint.annotation.PayloadRoot;
 import org.springframework.ws.server.endpoint.annotation.RequestPayload;
 import org.springframework.ws.server.endpoint.annotation.ResponsePayload;
-
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringManagementEndpoint.java:71]: the empty line was fine here, roll back the change

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/exceptions/ValidationException.java [hunk 22]
```diff
@@ -19,11 +19,28 @@
 @SoapFault(faultCode = FaultCode.SERVER)
 public class ValidationException extends PlatformException {
 
+    private static final String DEFAULT_MESSAGE = "Validation Exception";
     /**
      * Serial Version UID.
      */
     private static final long serialVersionUID = 9063383618380310347L;
-    private static final String DEFAULT_MESSAGE = "Validation Exception";
+
```
Comment (robindenadel) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/exceptions/ValidationException.java:27]: why was the order of methods changed?

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/ClockStatus.java [hunk 23]
```diff
@@ -53,7 +54,7 @@ public boolean hasStatusBits() {
 
     public Set<ClockStatusBit> getStatusBits() {
         if (this.statusBits == null) {
-            return null;
+            return new HashSet<ClockStatusBit>();
```
Comment (robindenadel) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/ClockStatus.java:57]: this line introduced another sonar issue

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/ClockStatusBit.java [hunk 24]
```diff
@@ -10,41 +10,30 @@
...
-    DOUBTFUL_VALUE("doubtful value"),
+    DAYLIGHT_SAVING_ACTIVE("daylight saving active"),
     DIFFERENT_CLOCK_BASE("different clock base"),
+    DOUBTFUL_VALUE("doubtful value"),
     INVALID_CLOCK_STATUS("invalid clock status"),
+    INVALID_VALUE("invalid value"),
     RESERVED_1("reserved"),
     RESERVED_2("reserved"),
-    RESERVED_3("reserved"),
-    DAYLIGHT_SAVING_ACTIVE("daylight saving active");
```
Comment (robindenadel) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/ClockStatusBit.java:22]: why was the order of elements changed?

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/ClockStatusBit.java [hunk 25]
```diff
@@ -10,41 +10,30 @@
...
-
-    public boolean isSet(final int clockStatus) {
-        final int mask = 1 << this.ordinal();
-        return mask == (mask & clockStatus);
-    }
-
-    @Override
-    public String toString() {
-        return this.description;
-    }
```
Comment (robindenadel) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/ClockStatusBit.java:42]: why was the order of methods changed?

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/ClockStatusBit.java [hunk 26]
```diff
@@ -10,41 +10,30 @@
...
     public static Set<ClockStatusBit> forClockStatus(final byte clockStatus) {
         return forClockStatus(clockStatus & 0xFF);
     }
 
+    // SonarQube complains about the null return. SonarQube wants to see a
+    // return of an empty set. The problem is that the behavior of the code
+    // differs depending on whether the return value is null or an empty set.
+    // Trying to correct this would result in other behavior changes, making a
+    // simple fix not possible. It is still an issue though, meaning it
+    // shouldn't be suppressed.
```
Comment (robindenadel) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/ClockStatusBit.java:36]: very good!

**Review by robindenadel — CHANGES_REQUESTED** [review]

File: osgp/platform/osgp-adapter-domain-smartmetering/src/test/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/mapping/customconverters/PushSetupAlarmDtoConverterTest.java [hunk 27]
```diff
@@ -0,0 +1,68 @@
...
+import org.opensmartgridplatform.domain.core.valueobjects.smartmetering.PushSetupAlarm;
+import org.opensmartgridplatform.domain.core.valueobjects.smartmetering.WindowElement;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.PushSetupAlarmDto;
+
+@ExtendWith(MockitoExtension.class)
+class PushSetupAlarmDtoConverterTest {
+    @Mock
+    private ConfigurationMapper configurationMapper;
+    @Mock
+    private PushSetupAlarmDto pushSetupAlarmDto;
```
Comment (MithraicMagic) [inline osgp/platform/osgp-adapter-domain-smartmetering/src/test/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/mapping/customconverters/PushSetupAlarmDtoConverterTest.java:28]: Mock is not used in class. Maybe remove?

File: osgp/platform/osgp-adapter-domain-smartmetering/src/test/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/mapping/customconverters/PushSetupAlarmDtoConverterTest.java [hunk 28]
```diff
@@ -0,0 +1,68 @@
...
+    @Mock
+    private MappingContext mappingContext;
+    @Mock
+    private PushSetupAlarm pushSetupAlarm;
+    private PushSetupAlarmDtoConverter pushSetupAlarmDtoConverter;
+
+    @Test
+    void convertTest() {
+        this.pushSetupAlarmDtoConverter = new PushSetupAlarmDtoConverter(this.configurationMapper);
```
Comment (MithraicMagic) [inline osgp/platform/osgp-adapter-domain-smartmetering/src/test/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/mapping/customconverters/PushSetupAlarmDtoConverterTest.java:37]: This is used in both functions, set globally in class?

**Review by MithraicMagic — CHANGES_REQUESTED** [review]

**Review by MithraicMagic — APPROVED** [review]
Looks good!

File: osgp/platform/osgp-adapter-domain-smartmetering/src/test/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/mapping/customconverters/PushSetupAlarmDtoConverterTest.java [hunk 29]
```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2020 Alliander N.V.
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-domain-smartmetering/src/test/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/mapping/customconverters/PushSetupAlarmDtoConverterTest.java:2]: 2021

File: osgp/platform/osgp-adapter-ws-core/src/test/java/org/opensmartgridplatform/adapter/ws/core/application/services/DeviceManagementServiceTest.java [hunk 30]
```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright 2020 Alliander N.V.
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-ws-core/src/test/java/org/opensmartgridplatform/adapter/ws/core/application/services/DeviceManagementServiceTest.java:2]: 2021

**Review by robindenadel — CHANGES_REQUESTED** [review]

**Comment by mvanOsch** [comment]
retest this please

**Review by robindenadel — APPROVED** [review]

**Pull request merged**